src/components/connectAdvanced.js

```diff
@@ -20,6 +22,13 @@ const stringifyComponent = Comp => {
   }
 }
 
+function storeStateUpdatesReducer(state, action) {
```

Comment:
i've hoisted this as this didnt depend on any closure variables, so should be better to reuse a single function for this rather than to create a copy for each connect